packages/toolpad-app/src/server/data.ts

```diff
@@ -233,6 +233,21 @@ export async function updateApp(appId: string, name: string): Promise<App> {
   });
 }
 
+export async function duplicateApp(appId: string): Promise<App> {
+  const app = await prisma.app.findUnique({ where: { id: appId } });
+  const dom = await loadPreviewDom(appId);
+  const duplicateDom = appDom.duplicateDom(dom);
+  if (!app) {
```

Comment:
not a big deal, but this check could be right after `findUnique` - just makes it a bit easier to follow and avoids doing unnecessary work.